File: Python/Path Sum/Path Sum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def hasPathSum(self, root, sum):
        """
        :type root: TreeNode
        :type sum: int
        :rtype: bool
        """
        if(not root):
            return False
        if(sum == root.val not root.left and not root.right):
            return True
        sum -= root.val
        leftHas = self.hasPathSum(root.left, sum)
        rightHas = self.hasPathSum(root.right, sum)
        return leftHas or rightHas

File: C#/Path Sum/Path Sum.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public bool HasPathSum (TreeNode root, int sum) {
        if (root == null) return false;
        if (sum == root.val && root.left == null && root.right == null) return true;
        sum -= root.val;
        bool leftHas = HasPathSum (root.left, sum);
        bool rightHas = HasPathSum (root.right, sum);
        return leftHas || rightHas;
    }
}